### PR TITLE
feat(fiberzap): Added 'FieldsFunc' to provide custom log fields

### PR DESCRIPTION
**Pull request by Jamess-Lucass**
Solves #750

### CODE REVIEW

File: fiberzap/zap_test.go [hunk 1]
```diff
@@ -441,3 +441,26 @@ func Test_LoggerLevelsAndMessagesSingle(t *testing.T) {
...
+
+	app.Get("/", func(c *fiber.Ctx) error {
+		return c.SendString("hello")
+	})
+
+	resp, err := app.Test(httptest.NewRequest("GET", "/", nil))
+
+	utils.AssertEqual(t, nil, err)
+	utils.AssertEqual(t, fiber.StatusOK, resp.StatusCode)
+	utils.AssertEqual(t, "test", logs.All()[0].Context[0].String)
```
Comment (gaby) [inline fiberzap/zap_test.go:465]: Can you add something like this to verify all the other expected fields are preset:

```go
expected := map[string]interface{}{
		"body":          "",
		"ip":            "0.0.0.0",
		"host":          "example.com",
		"url":           "/?foo=bar",
		"method":        "GET",
		"route":         "/",
		"protocol":      "http",
		"pid":           strconv.Itoa(os.Getpid()),
		"queryParams":   "foo=bar",
		"resBody":       "Cannot GET /",
		"bytesReceived": int64(0),
		"bytesSent":     int64(12),
	}
	utils.AssertEqual(t, expected, logs.All()[0].ContextMap())
```

**Review by gaby — CHANGES_REQUESTED** [review]

File: fiberzap/zap_test.go (hunk 1 above)
Comment (Jamess-Lucass) [reply fiberzap/zap_test.go]: See latest commit, took inspiration from `Test_Logger_All`

**Review by gaby — APPROVED** [review]

File: fiberzap/zap.go [hunk 2]
```diff
@@ -110,6 +110,11 @@ func New(config ...Config) fiber.Handler {
 
 		// Add fields
 		fields := make([]zap.Field, 0, len(cfg.Fields)+1)
+
+		if cfg.FieldsFunc != nil {
+			fields = cfg.FieldsFunc(c)
```
Comment (efectn) [inline fiberzap/zap.go:115]: Appending would be better

**Comment by Jamess-Lucass** [comment]
Thanks everyone <3

**Pull request merged**